### PR TITLE
chore: adding probing-vits

### PR DESCRIPTION
**Pull request by sayakpaul**

### CODE REVIEW

File: examples/vision/probing_vits.py [hunk 1]
```diff
@@ -0,0 +1,548 @@
+"""
+Title: Under the Hood of Vision Transformers
+Author: [Sayak Paul](https://twitter.com/RisingSayak), [Aritra Roy Gosthipaty](https://twitter.com/ariG23498)
```
Comment (sayakpaul) [inline examples/vision/probing_vits.py:3]: Should be written alphabetically with a note on equal contribution. Also it should be "Authors".

File: examples/vision/probing_vits.py [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+* calculate their geometric distance, and
+* multiply that with the attention scores.
+
+Attention scores are calculated here after forward passing the image in inference mode
+through the network. The following infographic may help you 
+understand the process a little bit better. 
+
+<img src="https://i.imgur.com/pZCgPwl.gif" height=500>
+
+This animation is created by [Ritwik Raha](https://twitter.com/ritwik_raha)
```
Comment (sayakpaul) [inline examples/vision/probing_vits.py:225]: "." missing.

File: examples/vision/probing_vits.py [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+ax2.axis("off")
+
+fig.tight_layout()
+fig.subplots_adjust(top=1.35)
+fig.show()
+
+"""
+You're encouraged to apply this method on the other models we mentioned and compare the
+results.
+"""
```
Comment (sayakpaul) [inline examples/vision/probing_vits.py:428]: Could you add a sentence describing what the reader is supposed to notice in the plot?

**Review by sayakpaul — COMMENTED** [review]
Minor changes are needed and then we should be good to proceed.

File: examples/vision/probing_vits.py (hunk 3 above)
Comment (ariG23498) [reply examples/vision/probing_vits.py:428]: Here do you mean a sentence like so:

You're encouraged to apply this method on the other models we mentioned and compare the results. The attention rollout plots will differ according to the tasks and augmentation the model was trained on. We observe that DeiT has
the best rollout plot due to its augmentation regieme.

Comment (sayakpaul) [reply examples/vision/probing_vits.py:428]: No I mean something like:

> We can notice that the model is able to focus its attention on the salient parts of the input image. You're encouraged to apply this method on the other models we mentioned and compare the results. The attention rollout plots will differ according to the tasks and augmentation the model was trained with. We observer that DeiT has the best rollout plot likely due to its augmentation regime.

**Comment by ariG23498** [comment]
I have made the necessary changes. Could you review and merge if everything seems right?

**Comment by sayakpaul** [comment]
@ariG23498 changes are a go. Feel free to open a PR to the keras-io master.

**Comment by ariG23498** [comment]
Aye aye cap'n

**Pull request merged**